invenio_drafts_resources/auditlog_actions.py

```diff
@@ -8,30 +8,104 @@
 
 """Action registration via entrypoint function."""
 
-from invenio_audit_logs.actions import AuditAction
-
-
-def record_actions():
-    """Function to add actions to the registry."""
-    return {
-        "draft.create": AuditAction(
-            name="draft.create",
-            message_template="User {user_id} created the draft {resource_id}.",
-        ),
-        "draft.edit": AuditAction(
-            name="draft.edit",
-            message_template="User {user_id} updated the draft {resource_id}.",
-        ),
-        "record.publish": AuditAction(
-            name="record.publish",
-            message_template="User {user_id} published the record {resource_id}.",
-        ),
-        "draft.delete": AuditAction(
-            name="draft.delete",
-            message_template="User {user_id} deleted the draft {resource_id}.",
-        ),
-        "draft.new_version": AuditAction(
-            name="draft.new_version",
-            message_template="User {user_id} created a new version {resource_id}.",
-        ),
-    }
+from invenio_access.permissions import system_identity
+from invenio_audit_logs.services import AuditLogBuilder
+from invenio_records.dictutils import dict_lookup, dict_set
+from invenio_records_resources.references.entity_resolvers import ServiceResultResolver
+
+
+class UserResolve:
+    """Payload generator for audit log using the service result resolvers."""
+
+    def __init__(self, key):
+        """Ctor."""
+        self.key = key
+
+    def __call__(self, resolver, log_data):
+        """Update required recipient information and add backend id."""
+        entity_ref = dict_lookup(log_data, self.key)
+        if entity_ref == system_identity.id:
+            entity_data = {
+                "id": str(system_identity.id),
+                "name": "system",
+                "email": "system@system.org",
+            }
+        else:
+            entity = resolver.get_entity_proxy({self.key: entity_ref}).resolve()
+            entity_data = {
+                "id": str(entity["id"]),
+                "name": entity["username"],
+                "email": entity["email"],
+            }
+        dict_set(log_data, self.key, entity_data)
+
+
+class RecordBaseAuditLog(AuditLogBuilder):
+    """Base class for audit log builders."""
+
+    context = [
+        {
+            "resolver": ServiceResultResolver(service_id="users", type_key="user"),
+            "generator": UserResolve(key="user"),
+        },
+    ]
```

Comment:
What about a more unified approach with objects for extracting/building the context:

```python
class RecordAuditLog(AuditLog):

    context = [
        UserContext(),
        RequestContext(),
        TaskContext(),
    ]
    
    ...
```